Resources/Prototypes/_Floof/Entities/Mobs/NPCs/pets.yml

```
@@ -18,3 +18,58 @@
     tags:
     - CannotSuicide
     - VimPilot
+
+- type: entity
+  name: Pet Based
+  parent: [MobGiantSpider, BaseItem]
+  id: MobPetBase
```

Cleanup: proper name and ID
```suggestion
  name: pet spider
  parent: [MobGiantSpider, BaseItem]
  id: MobPetSpiderBase
```